server/controllers/auth/signup.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable camelcase */
+const { hash } = require('bcrypt');
+const { sign } = require('jsonwebtoken');
+
+const { signUpClient } = require('../../database/queries/auth');
+const { getClient_Data } = require('../../database/queries/profile');
+
+const signUp = async (req, res, next) => {
+  const {
+    name,
+    password,
+    mobile_number,
+    facebook_profile,
+    location,
+    address,
+    avatar,
+  } = req.body;
+
+  // get random color to use it in avatar
+  const randomColor = Math.random().toString(16).substr(-6);
+  // generation avatar with client name
+  const defaultAvatar = `https://ui-avatars.com/api/?name=${name}&rounded=true&background=${randomColor}&color=F3F3F3`;
+  const favorite = [];
+
+  // check if mobile number or facebook is exist
+  const checkMobileNumber = await getClient_Data({ mobile_number });
+  const checkFaceBook = await getClient_Data({ facebook_profile });
+
+  if (checkFaceBook.rowCount > 0 || checkMobileNumber.rowCount > 0) {
+    res.status(400).json({ message: 'this account is already exist' });
+  } else {
+    // hash password
+    const hashedPassword = await hash(password, 10);
```

Comment:
in Facebook signup there is no password try to generate a password and hash it or try name and id . or try something else.